### PR TITLE
Add a simple way to check the type of EquipmentSlot

### PR DESCRIPTION
**Pull request by SoSeDiK**
Adds `EquipmentSlot#isHand` and `EquipmentSlot#isArmor` to check the type of the slot.

### CODE REVIEW

File: patches/api/0403-Add-LivingEntity-swingHand-EquipmentSlot-convenience.patch [hunk 1]
```diff
@@ -23,7 +23,7 @@ index 4af5e8d0cba6555f7615e4e809d9aff221c0dc4d..319e4571aca24d1e3f6c85b7435d65c0
 +     * @throws IllegalArgumentException if invalid hand is passed
 +     */
 +    default void swingHand(@NotNull org.bukkit.inventory.EquipmentSlot hand) {
-+        com.google.common.base.Preconditions.checkArgument(hand == org.bukkit.inventory.EquipmentSlot.HAND || hand == org.bukkit.inventory.EquipmentSlot.OFF_HAND, String.format("Expected a valid hand, got \"%s\" instead!", hand));
++        com.google.common.base.Preconditions.checkArgument(hand.isHand(), String.format("Expected a valid hand, got \"%s\" instead!", hand));
```
Comment (jpenilla) [inline patches/api/0403-Add-LivingEntity-swingHand-EquipmentSlot-convenience.patch]: potential npe
```suggestion
+        com.google.common.base.Preconditions.checkArgument(hand != null && hand.isHand(), String.format("Expected a valid hand, got \"%s\" instead!", hand));
```

File: patches/api/0402-Add-EquipmentSlot-convenience-methods.patch [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+     LEGS,
+     CHEST,
+     HEAD
++    // Paper start
++    ;
++    /**
++     * Checks whether this equipment slot is a hand:
++     * either {@link #HAND} or {@link #OFF_HAND}
++     *
++     * @return true if this is a hand slot
```
Comment (jpenilla) [inline patches/api/0402-Add-EquipmentSlot-convenience-methods.patch]:
```suggestion
+     * @return whether this is a hand slot
```

File: patches/api/0402-Add-EquipmentSlot-convenience-methods.patch [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
++        return this == HAND || this == OFF_HAND;
++    }
++
++    /**
++     * Checks whether this equipment slot
++     * is one of the armor slots:
++     * {@link #HEAD}, {@link #CHEST},
++     * {@link #LEGS}, or {@link #FEET}
++     *
++     * @return true if this is an armor slot
```
Comment (jpenilla) [inline patches/api/0402-Add-EquipmentSlot-convenience-methods.patch]:
```suggestion
+     * @return whether this is an armor slot
```

**Review by jpenilla — CHANGES_REQUESTED** [review]

**Review by kennytv — APPROVED** [review]

**Review by YanisBft — APPROVED** [review]

**Review by Machine-Maker — APPROVED** [review]

**Review by jpenilla — APPROVED** [review]

**Comment by theminecoder** [comment]
Will the fact that the patch that adds the methods is added after the patch that uses it affect version upgrades in the future (when likely applying patches one by one)?

**Comment by Owen1212055** [comment]
I mean, because it's just one API method it most likely won't have any effect. However, it's probably better if those patches are moved. I'll go ahead and rename that in the PR.

**Pull request merged**